sui-execution/latest/sui-adapter/src/execution_engine.rs

```diff
@@ -1027,10 +1032,12 @@ mod checked {
             }
         }
 
-        if protocol_config.fresh_vm_on_framework_upgrade() {
+        let incoming_protocol_config =
+            ProtocolConfig::get_for_version(change_epoch.protocol_version, temporary_store.chain());
+        if incoming_protocol_config.fresh_vm_on_framework_upgrade() {
```

Comment:
Note: this is important to support large protocol version jumps -- in particular jumps that may jump from before this change (e.g., protocol version 40 or something that doesn't have this flag) to far after this change (e.g., version 85) when we in fact do want to use the updated binary config in that case.